pool/src/lib.rs

```diff
@@ -530,25 +525,76 @@ impl LendingPool {
     fn do_deposit(
         env: &Env,
         who: &Address,
-        s_token_address: &Address,
         asset: &Address,
         amount: i128,
-        collat_accrued_rate: i128,
+        reserve: &ReserveData,
     ) -> Result<bool, Error> {
-        let token = token::Client::new(env, asset);
-        token.transfer(who, s_token_address, &amount);
+        if amount == 0 {
+            return Ok(false);
+        }
 
-        let s_token = s_token_interface::STokenClient::new(env, s_token_address);
+        let collat_coeff = Self::get_collateral_coeff(env, reserve)?;
+        let underlying_asset = token::Client::new(env, asset);
+        let s_token = STokenClient::new(env, &reserve.s_token_address);
         let is_first_deposit = s_token.balance(who) == 0;
 
-        // amount_to_mint = amount / collat_accrued_rate
-        let amount_to_mint = FixedI128::from_inner(collat_accrued_rate)
+        let amount_to_mint = collat_coeff
             .recip_mul_int(amount)
             .ok_or(Error::MathOverflowError)?;
+
+        underlying_asset.transfer(who, &reserve.s_token_address, &amount);
         s_token.mint(who, &amount_to_mint);
+
+        event::deposit(env, who.clone(), asset.clone(), amount);
+
         Ok(is_first_deposit)
     }
 
+    fn do_repay(
```

Comment:
Could you add comment what exactly the method returns. It is not clear the meanings of pair (i128, bool)